### PR TITLE
wip_Features/implement authorization and authentication flow

### PR DESCRIPTION
**Pull request by pako535**

### CODE REVIEW

File: views/index.py [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+                    session['id'] = data.id
+                    session['email'] = data.email
+                    print(session)
+                    # load_user(data.email)
+                    return render_template('index.html')
+                else:
+                    return 'Login failed'
+            else:
+                return 'Incorrect data'
+        except:
```
Comment (etude-ist) [inline views/index.py]: Good practice is to specify what kind of exception occurs. Let's fix it.

File: views/index.py [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
+    else:
+        form = LoginForm()
+        try:
+            if form.validate():
+                data = User.query.filter_by(email=form.email.data).first()
+                if data is not None and check_password_hash(data.password_hash, form.password.data):
+                    session['logged_in'] = True
+                    session['id'] = data.id
+                    session['email'] = data.email
+                    print(session)
```
Comment (etude-ist) [inline views/index.py]: Eliminate print from source code. To debug better use pdb: [https://docs.python.org/3/library/pdb.html](url)

File: views/index.py [hunk 3]
```diff
@@ -1,7 +1,65 @@
...
+                    # load_user(data.email)
+                    return render_template('index.html')
+                else:
+                    return 'Login failed'
```
Comment (etude-ist) [inline views/index.py]: This should be something more sane. Such info should be passed to form and be rendered.

File: views/index.py [hunk 4]
```diff
@@ -1,7 +1,65 @@
...
+        return render_template('registration.html', form=form)
+    else:
+        form = RegistrationForm()
+        if form.validate():
+            try:
+                new_user = User(email=form.email.data, first_name=form.first_name.data, surname=form.surname.data,
+                                password_hash=generate_password_hash(form.password.data))
+                db.session.add(new_user)
+                db.session.commit()
+            except:
```
Comment (etude-ist) [inline views/index.py]: Same as previous comment.

File: views/index.py [hunk 5]
```diff
@@ -1,7 +1,65 @@
-from flask import render_template
+from flask import render_template, request, session
 from . import library
+from models.user import User
+from forms.login_form import LoginForm
+from forms.registration_forms import RegistrationForm
+from werkzeug.security import generate_password_hash, check_password_hash
+from app import db
+from flask_login import LoginManager
```
Comment (etude-ist) [inline views/index.py]: Let's order, group properly imports, first python standard library, then python libraries, then flask code, then app code. Fix that also in the rest of your submitted changes.

File: .pytest_cache/v/cache/lastfailed [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (pako535) [reply .pytest_cache/v/cache/lastfailed]: During merging added, and it comes not from my branch, but i fix it in Wednesday

File: config.py [hunk 7]
```diff
@@ -16,6 +16,7 @@ class DevConfig(Config):
     DB_HOST = getenv('DB_HOST')
     DB_PORT = getenv('DB_PORT')
     DB_NAME = getenv('DB_NAME')
+    SECRET_KEY = getenv('SECRET_KEY') or 'blaah'
```
[the remark this answers is not in the document]
Comment (pako535) [reply config.py]: it comes not from my branch, but i fix it in Wednesday 
for this he was responsible Łukasz and Liza

File: config.py [hunk 8]
```diff
@@ -25,7 +26,14 @@ class DevConfig(Config):
         DB_PORT,
         DB_NAME,
     )
-
+    # email server
+    MAIL_SERVER = 'smtp.gmail.com'
+    MAIL_PORT = 465
+    MAIL_USE_TLS = False
+    MAIL_USE_SSL = True
+    MAIL_USERNAME = 'tieto.library@gmail.com'
```
[the remark this answers is not in the document]
Comment (pako535) [reply config.py]: it comes not from my branch, but i fix it in Wednesday 
for this he was responsible Łukasz and Liza

File: forms/forms.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    send_message = SubmitField('Send message')
+
+
+class RegistrationForm(FlaskForm):
+    email = StringField('Email', validators=[Email()])
+    first_name = StringField('First name', validators=[DataRequired()])
+    surname = StringField('Surname', validators=[DataRequired()])
+    password = PasswordField('Password', validators=[DataRequired(),
+                                                     EqualTo('confirmPass', message='Passwords must match.')])
+    confirmPass = PasswordField('Confirm password', validators=[DataRequired()])
```
[the remark this answers is not in the document]
Comment (pako535) [reply forms/forms.py]: it comes not from my branch, but i fix it in Wednesday 
for this he was responsible Łukasz

**Pull request closed without merging**